android/app/src/main/java/io/peggyjo/peggymeter/MainActivity.java

```diff
@@ -58,9 +58,8 @@ private void showStatScreen() {
         }
         switch (currentMode) {
             case Graph:
-                if (historyGraphFragment == null) {
-                    historyGraphFragment = new HistoryGraphFragment();
-                }
+                dataController.clearMoodListeners();
+                historyGraphFragment = new HistoryGraphFragment();
```

Comment:
Can you add a comment explaining why we re-create this fragment every time this function is called please?

Reply:
done